rest-service/manager_rest/rest/resources_v3_1/execution_schedules.py

```diff
@@ -0,0 +1,275 @@
+                count = convert_to_int(request_dict.get('count'))
+            frequency = self._verify_frequency(request_dict.get('frequency'))
+            weekdays = self._verify_weekdays(request_dict.get('weekdays'))
+            if not frequency and count != 1:
+                raise manager_exceptions.BadParametersError(
+                    "frequency must be specified for execution count larger "
+                    "than 1")
+            rule = {
+                'frequency': frequency,
+                'count': count,
+                'weekdays': weekdays
+            }
+
+        slip = request_dict['slip']
+        stop_on_fail = verify_and_convert_bool(
+            'stop_on_fail',  request_dict.get('stop_on_fail', False))
+
+        now = get_formatted_timestamp()
+        schedule = models.ExecutionSchedule(
+            id=schedule_id,
+            deployment=deployment,
+            created_at=now,
+            since=since,
+            until=until,
+            rule=rule,
+            slip=slip,
+            workflow_id=workflow_id,
+            parameters=parameters,
+            execution_arguments=execution_arguments,
+            stop_on_fail=stop_on_fail,
+        )
+        return rm.sm.put(schedule), 201
+
+    @rest_decorators.marshal_with(models.ExecutionSchedule)
+    @authorize('execution_start')
+    def patch(self, schedule_id, **kwargs):
+        """Updates scheduling parameters of an existing execution schedule"""
+
+        validate_inputs({'schedule_id': schedule_id})
+        request_dict = get_json_and_verify_params()
+
+        sm = get_storage_manager()
+        schedule = sm.get(models.ExecutionSchedule, schedule_id)
+        since = self._verify_timestamp(request_dict.get('since'))
+        until = self._verify_timestamp(request_dict.get('until'))
+        slip = request_dict.get('slip')
+        stop_on_fail = request_dict.get('stop_on_fail')
+
+        if since:
+            schedule.since = since
+        if until:
+            schedule.until = until
+        if slip is not None:
+            schedule.slip = slip
+        if stop_on_fail is not None:
+            schedule.stop_on_fail = verify_and_convert_bool('stop_on_fail',
+                                                            stop_on_fail)
+
+        rrule_string = request_dict.get('rrule')
+        frequency = request_dict.get('frequency')
+        weekdays = request_dict.get('weekdays')
+        count = request_dict.get('count')
+
+        if rrule_string:
+            if frequency or weekdays or count:
+                raise manager_exceptions.BadParametersError(
+                    "`rrule` cannot be provided together with `frequency`, "
+                    "`weekdays` or `count`.")
+            try:
+                rrule.rrulestr(rrule_string)
+            except ValueError as e:
+                raise manager_exceptions.BadParametersError(
+                    "invalid RRULE string provided: {}".format(str(e)))
```

Comment:
```suggestion
                    "invalid RRULE string provided: {}".format(e))
```
str.format already calls str() on the arguments